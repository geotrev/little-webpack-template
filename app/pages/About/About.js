import React from "react"
import PropTypes from "prop-types"
import ScrollUpOnMount from "app/helpers/ScrollUpOnMount"

export default function About({ message }) {
  return (
    <div id="about">
      <ScrollUpOnMount />
      <span role="img" aria-label="thumbs-up emoji" className="hopping emoji">
        üëç
      </span>
      <h1>{"You found another page!"}</h1>
      <p>{message}</p>
    </div>
  )
}

About.propTypes = {
  message: PropTypes.string,
}
