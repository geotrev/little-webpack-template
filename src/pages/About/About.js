import React from "react"
import { ScrollUpOnMount } from "helpers"

const About = () => (
  <div id="about">
    <ScrollUpOnMount />
    <div className="hopping emoji">👍</div>
    <h1>You found an About page!</h1>
    <p>Woohoo, you did it.</p>
  </div>
)

export default About
