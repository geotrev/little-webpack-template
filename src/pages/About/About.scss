#about {
  flex: 1 1 auto;
  max-width: 100%;
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  font-family: monospace;
  font-size: 16px;
  text-align: center;
  padding: 120px 20px;

  .hopping {
    animation: hop 1s alternate infinite;
  }
}

@keyframes hop {
  0% {
    margin: -15px 0 15px 0;
  }

  80%, 100% {
    margin: 0;
  }

  10% {
    margin: -15px 0 15px 0;
  }
}
