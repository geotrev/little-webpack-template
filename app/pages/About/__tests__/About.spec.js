import React from "react"
import About from "../About"

jest.mock("app/helpers/ScrollUpOnMount", () => global.simpleMock("ScrollUpOnMount"))

describe("<About />", () => {
  it("renders", () => {
    const wrapper = mount(<About />)
    expect(wrapper).toMatchSnapshot()
  })
})
