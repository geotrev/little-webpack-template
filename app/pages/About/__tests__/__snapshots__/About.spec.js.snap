// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> renders 1`] = `
<About>
  <div
    id="about"
  >
    <ScrollUpOnMount />
    <span
      aria-label="thumbs-up emoji"
      className="hopping emoji"
      role="img"
    >
      👍
    </span>
    <h1>
      You found another page!
    </h1>
    <p />
  </div>
</About>
`;
